### PR TITLE
Modify index-pbench for pbench-user-benchmark

### PR DESCRIPTION
**Pull request by ndokos**
Deal with pbench-user-benchmark peculiarities:
    
    - There are no samples, just a reference-result.
    - The iteration name is just "1" with no dashes.

### CODE REVIEW

File: server/pbench/bin/index-pbench [hunk 1]
```diff
@@ -675,19 +678,23 @@ def get_samples(iteration, mdlog, tb):
...
+    if len(samples) == 0:
+        samples.append('reference-result')
     return samples
 
 def get_pbench_hosts(iteration, sample, mdlog, tb):
     try:
         # N.B. Space-separated list
         return mdlog.get("tools", "hosts").split(' ')
-    except Exception:
+    except Exception as e:
```
Comment (portante) [inline server/pbench/bin/index-pbench]: Catching `Exception` is very broad, what about catching [configparser.Error](https://docs.python.org/3/library/configparser.html#configparser.Error) instead?

**Comment by ndokos** [comment]
Working on unit tests.

File: server/pbench/bin/index-pbench [hunk 2]
```diff
@@ -675,19 +678,37 @@ def get_samples(iteration, mdlog, tb):
...
         # N.B. Space-separated list
         return mdlog.get("tools", "hosts").split(' ')
-    except Exception:
+    except ConfigParserError:
+        print("ConfigParser error in get_pbench_hosts: tool data will *not* be indexed.")
+        print("This is most probably a bug: please open an issue.")
+        return []
+    except NoSectionError:
+        print("No [tools] section in metadata.log: tool data will *not* be indexed.")
+        retun []
```
Comment (portante) [inline server/pbench/bin/index-pbench]: s/retun/return

File: server/pbench/bin/index-pbench [hunk 3]
```diff
@@ -675,19 +678,37 @@ def get_samples(iteration, mdlog, tb):
...
     try:
         return mdlog.options("tools/{}".format(host))
-    except Exception:
+    except ConfigParserError:
+        print("ConfigParser error in get_tools: tool data will *not* be indexed.")
+        print("This is most probably a bug: please open an issue.")
+        return []
+    except NoSectionError:
+        print("No [tools/{}] section in metadata.log: tool data will *not* be indexed.".format(host))
+        retun []
```
Comment (portante) [inline server/pbench/bin/index-pbench]: s/retun/return

**Pull request merged**